reference/commands/consumer/install.rst

```diff
@@ -333,6 +330,22 @@ because the consumer conanfile might not declare a `name` so it would be impossi
     You can use :ref:`profiles <profiles>` files to create predefined sets of **settings**,
     **options** and **environment variables**.
 
+folders
+-------
+
+.. warning::
+
+    This is an **experimental** feature subject to breaking changes in future releases.
+
+
+The ``--output-folder`` and ``--source-folder`` define together with the ``layout()`` recipe
+method the location of the output files. For example, the files created by build system integrations
+such as ``CMakeToolchain`` or ``PkgConfigDeps`` will be created in the folder defined by the
+``layout()`` ``generators`` folder, inside the defined ``--output-folder``. By default, both the
+``--output-folder`` and the ``--source-folder`` are the folder containing the ``conanfile.py`` or
+the current folder otherwise.
```

Comment:
Maybe we can add an example or something? I think this part is a bit hard to understand